src/Sensor/SensorBms.cpp

```diff
@@ -0,0 +1,144 @@
+            _canInterface.sendMessage(msg);
+        }
+        
+        _lastValidTime = millis();
+    }
+}
+
+String SensorBms::getHumanName() {
+    return "BMS";
+}
+
+void SensorBms::update(CanMessage message) {
+    DEBUG_SERIAL_LN("RECEIVED RESPONSE FROM BMS");
+
+    if(message.data[RSP_STATUS_BYTE] != TRUE) {
+        DEBUG_SERIAL_LN("Poor Data Received");
+    }
+    else {
+        unsigned statusCode = 0;
+        switch (message.data[RSP_PARAM_ID_BYTE]) {
+            case PARAM_ID_BATTERY_VOLTAGE:
+                _batteryVoltage = parseFloat(message.data + RSP_DATA_BYTE);
+                break;
+            case PARAM_ID_BATTERY_CURRENT:
+                _batteryCurrent = parseFloat(message.data + RSP_DATA_BYTE);
+                break;
+            case PARAM_ID_MAX_CELL_VOLTAGE:
+                _cellVoltageMax = parseFloat(message.data + RSP_DATA_BYTE);    
+                break;
+            case PARAM_ID_MIN_CELL_VOLTAGE:
+                _cellVoltageMin = parseFloat(message.data + RSP_DATA_BYTE);    
+                break;
+            case PARAM_ID_STATUS:
+                statusCode = (message.data[3] << 8) + message.data[2];
+                if(statusCode == STATUS_CHARGING) {
+                    _bmsStatus = "BMS Charging...";
+                    DEBUG_SERIAL_LN("BMS Charging...");
+                }
+                else if(statusCode == STATUS_CHARGED) {
+                    _bmsStatus = "BMS Charged!";
+                    DEBUG_SERIAL_LN("BMS Charged!");
+                }
+                else if(statusCode == STATUS_DISCHARGING) {
+                    _bmsStatus = "BMS Discharging...";
+                    DEBUG_SERIAL_LN("BMS Discharging...");
+                }
+                else if(statusCode == STATUS_REGENERATION) {
+                    _bmsStatus = "BMS Regeneration";
+                    DEBUG_SERIAL_LN("BMS Regeneration");
+                }
+                else if(statusCode == STATUS_IDLE) {
+                    _bmsStatus = "BMS Idle";
+                    DEBUG_SERIAL_LN("BMS Idle");
+                }
+                else {
+                    _bmsStatus = "BMS Fault Error";
+                    DEBUG_SERIAL_LN("BMS Fault Error");
+                }
+                break;
+            case PARAM_ID_SOC:
+                _soc = parseFloat(message.data + RSP_DATA_BYTE);    
+                break;
+            case PARAM_ID_TEMP:
+                if(message.data[5] == TEMP_ID_INTERNAL) {
+                    _tempBms = (message.data[4] << 8) + message.data[3];
+                }
+                else if(message.data[5] == TEMP_ID_BATTERY_1) {
+                    _batteryTemp1 = (message.data[4] << 8) + message.data[3];
+                }
+                else {
+                    _batteryTemp2 = (message.data[4] << 8) + message.data[3];
+                }
+                break;
+            default:
+                break;
+        }
+    }
+}
+
+float SensorBms::getBatteryVolt() {
```

Comment:
Can you update these getters to return "String" instead of "float"? You can use the macro FLOAT_TO_STRING() that's defined in the settings.h file to specify the number of decimal points. The sensor spreadsheet on notion contains all the decimal points for each sensor parameter.

Reply:
All done. I left the temperature sensors as ints since the number is fine as it is. Should I convert them too?

Reply:
Yes, let's leave ints as ints because we don't need to specify the number of decimal places